hail/python/hailtop/aiotools/router_fs.py

```diff
@@ -55,10 +57,14 @@ def _load_fs(self, uri: str):
                 **self._gcs_kwargs,
                 bucket_allow_list = self._gcs_bucket_allow_list.copy()
             )
-        elif aioazure.AzureAsyncFS.valid_url(uri):
-            fs = aioazure.AzureAsyncFS(**self._azure_kwargs)
         elif aioaws.S3AsyncFS.valid_url(uri):
             fs = aioaws.S3AsyncFS(**self._s3_kwargs)
+        # Azure URLs are also valid for Terra on Azure
+        elif aioazure.AzureAsyncFS.valid_url(uri):
+            if aioterra_azure.TerraAzureAsyncFS.configured():
```

Comment:
I don't love configured? Maybe "enabled"? Or just "in_terra_azure"?

Reply:
Changed to enabled.